teps/0047-finallytask-execution-post-timeout.md

```diff
@@ -0,0 +1,265 @@
+updates.
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+This TEP adresses issue [`#2980`](https://github.com/tektoncd/pipeline/issues/2989). 
+
+The proposal is to enable finally tasks to execute when a pipelinerun has reached timeout.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this TEP.  Describe why the change is important and the benefits to users.  The
+motivation section can optionally provide links to [experience reports][] to
+demonstrate the interest in a TEP within the wider Tekton community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+The finally task [`design document`](https://docs.google.com/document/d/1lxpYQHppiWOxsn4arqbwAFDo4T0-LCqpNa6p-TJdHrw/edit#heading=h.w51ed6k2inef) list the following use cases :
+
+- Cleanup cluster resources after finishing (with success/failure) integration tests (Dogfooding Scenario)
+- Update Pull Request with what happened overall in the pipeline (pipeline level)
+- Report Test Results at the end of the test pipeline (Notifications Scenario)
+
+Unfortunately if a pipeline's execution reaches the defined timeout value, the pipelinerun stop and reports a failed status.
+
+Here is an example pipeline run with a finally task:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: hello-world-pipeline-run-with-timeout
+spec:
+  timeout: "0h0m60s"
+  pipelineSpec:
+    tasks:
+    - name: task1
+      timeout: "0h0m30s"
+      taskSpec:
+        steps:
+          - name: hello
+            image: ubuntu
+            script: |
+              echo "Hello World!"
+              sleep 10
+    finally:
+    - name: task2
+      params:
+        - name: echoStatus
+          value: "$(tasks.task1.status)"
+      taskSpec:
+        params:
+          - name: echoStatus
+        steps:
+          - name: verify-status
+            image: ubuntu
+            script: |
+              if [ $(params.echoStatus) == "Succeeded" ]
+              then
+                echo " Hello World echoed successfully"
+              fi
+```
+
+The finally task runs after the task completion and both execute normally.
+
+
+| NAME                                                           | TASK NAME        | STARTED        | DURATION   | STATUS                 |
+|----------------------------------------------------------------|------------------|----------------|------------|------------------------|
+| ∙ hello-world-pipeline-run-with-timeout-task1-tp9dq          | task1          | 17 seconds ago | 4 seconds  | Succeeded              |
+| ∙ hello-world-pipeline-run-with-timeout-task2-zpzfm | task2 | 47 seconds ago | 30 seconds | Succeeded |
+|                                                                |                  |                |            |                        |
```

Comment:
Not a blocker here, but this is weird:

* dag task `Task1` with `sleep 10` has Duration `4 seconds`
* final task `Task2` without any delay has Duration of `30 seconds`

Reply:
Thanks for reviewing. Will check this out